docs/metrics/getting-started/Program.cs

```diff
@@ -0,0 +1,120 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using OpenTelemetry;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Trace;
+
+public class Program
+{
+    private static readonly ActivitySource MyActivitySource = new ActivitySource(
+        "MyCompany.MyProduct.MyLibrary");
+
+    public static void Main()
+    {
+        using var tracerProvider = Sdk.CreateTracerProviderBuilder()
+            .AddSource("MyCompany.MyProduct.MyLibrary")
+            .AddConsoleExporter()
+            .Build();
+
+        /*
+        # Proposed Design Principles
+
+        * The API should be easy to use - for both application and library developers.
+          * Instrumentation API - e.g. report an int number with few dimensions.
+          * Configuration API - e.g. get the meter, configure the aggregation time
+            window.
+          * Consumption API - e.g. define a histogram.
+        * Metrics should be reliable.
+          * Not going to use unbounded resource.
+          * Any time we callback to user code, we shouldn't take lock (be lock-free).
+          * Metrics should be trust-worthy.
+          * Correctness and fidelity. It doesn't mean that all the metrics data should
+            be 100% accurate - sometimes from engineering perspective we might prefer an
+            approximate value if it yields great performance/efficiency benefit. The
+            minimum bar here is that the behavior should be well defined/documented and
+            backed by mathematics.
+          * In case the implementation needs to reduce data quality (e.g. throttle,
+            resource cap, integer overflow, etc.), there should be a way for the
+            consumer to know if the data should be trusted or not.
+        * Metrics should be performant.
+          * If most counter values could fit into 8-bit integer, we shouldn't start from
+            allocating 16-bit.
+          * All the hot-path should be lock-free - does not cause CPU context switch.
+            The actual definition of "what are hot-paths" is subjective and should be
+            hammered out as we started to build the PoC (for example, there could be
+            batch updates for multiple metric data points which could take a lock).
+        */
+
+        MeterProvider.SetDefault(Sdk.CreateMeterProviderBuilder()
```

Comment:
The code should be broken up into 2 separate sections.

1. The instrumentation code (which DOES NOT have SDK).
This side only calls GetMeter() without knowing which SDK it's tied to.

2. The "operator" side.
This is the app that will attach an SDK and make the meters "real".

Reply:
section2 started at Line 73.